modules/navier_stokes/src/fvkernels/PINSFVTurbulentDiffusion.C

```diff
@@ -0,0 +1,80 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PINSFVTurbulentDiffusion.h"
+#include "INSFVEnergyVariable.h"
+#include "NS.h"
+
+registerMooseObject("NavierStokesApp", PINSFVTurbulentDiffusion);
+
+InputParameters
+PINSFVTurbulentDiffusion::validParams()
+{
+  auto params = FVFluxKernel::validParams();
+  params.addClassDescription("Diffusion term in the porous media "
+                             "partially incompressible Navier-Stokes turbulence"
+                             "fluid energy equations :  $-div(eps * k / turb_coef * grad(T))$");
+  params.addRequiredParam<MooseFunctorName>("mu_t", "Thermal conductivity");
+  params.addRequiredParam<MooseFunctorName>(NS::porosity, "Porosity");
+  params.addRequiredParam<MooseFunctorName>("turb_coef", "Normalization Coefficient Turbulent Diffusion");
+  params.addParam<bool>(
+      "effective_diffusivity",
+      false,
+      "Whether the diffusivity should be multiplied by porosity, or whether the provided "
+      "diffusivity is an effective diffusivity taking porosity effects into account");
+
+  params.set<unsigned short>("ghost_layers") = 2;
+  return params;
+}
+
+PINSFVTurbulentDiffusion::PINSFVTurbulentDiffusion(const InputParameters & params)
```

Comment:
The comment is the same as before, would be cool to do turbulent and non-turbulent kernels together. In this case the only additional functionalities are the `turb_coeff` and the porosity switch, right?